src/front/glsl/error.rs

```diff
@@ -14,8 +22,8 @@ pub enum ErrorKind {
     InvalidProfile(SourceMetadata, String),
     #[error("Invalid version: {1}")]
     InvalidVersion(SourceMetadata, u64),
-    #[error("Unexpected token: {0}")]
-    InvalidToken(Token),
+    #[error("Expected {1}, got token '{0}'")]
+    InvalidToken(Token, Cow<'static, str>),
```

Comment:
We should store a `Vec` of the possible tokens here instead of a string and the message should be `Expected {0}, found {1}` since it's more compact and doesn't involve the user with implementation details like tokens

Reply:
Done.